src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -226,7 +226,7 @@ public JsonCommentHandling ReadCommentHandling
             set
             {
                 VerifyMutable();
-                if (value == JsonCommentHandling.Allow)
+                if (!JsonHelpers.IsCommentHandlingSupported(value, JsonCommentHandling.Skip))
```

Comment:
Thanks @GrabYourPitchforks for pointing out this issue to me.

Reply:
FWIW, I find this helper method very confusing.  It's not clear from its name that it's doing a range check based on the second argument.  That it's doing a range check means I need to know in my head the values associated with each enum value to know their associated order. I need to know that the range check is inclusive.  Etc.  If it's going to do a range check, it should just be named that: in fact, you could just get rid of the helper and use the underlying IsInRangeInclusive, which would at least be self-explanatory as to what it's doing.

Reply:
It also doesn't seem like any of these are hot-path enough to warrant the obfuscation that comes with this.  Wouldn't it be better to just do:
```C#
if (value != JsonCommentHandling.Disallow && value != JsonCommentHandling.Skip)
{
    throw ...;
}
```
?